src/Comps/dropdownsFilters/DropDowns.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useState } from "react";
+import DropDown from "../select/DropDown";
+import { DisplayRowWithGap } from "./style";
+import { RootState } from "../../store/store";
+import { useSelector, useDispatch } from "react-redux";
+import { updateFilters, resetFilters, updateSortBy } from "../../store/query";
+import {
+  SortbyData,
+  CategoryData,
+  CountryData,
+  Language,
+  LanguageCodesAndNames,
+  setToLanguageCode,
+  everythingFilters,
+  topHeadlinesFilters,
+} from "../../types/datatypes";
+import Dates from "./Dates";
+import { getAllTopHeadlinesSources } from "../../Services/Api";
+
+const DropDowns: React.FC = () => {
+  const [filters, setFilters] = useState<string[]>(topHeadlinesFilters);
+  const [sources, setSources] = useState<string[]>();
+  const Query = useSelector((state: RootState) => state.query);
+
+  useEffect(() => {
+    getAllTopHeadlinesSources(Query.query.filters.country).then((value) => {
+      setSources(value);
+    });
+  }, [Query.query.filters.country]);
+
+  //convert Countries and Languages code to names and names to code
+  var countries = require("i18n-iso-countries");
```

Comment:
will that function be used in other files in the project?
if yes, might be good to create a utils file that will contain reuse-able functions